lib/puma_worker_killer.rb

```diff
@@ -24,9 +28,12 @@ def start(frequency = self.frequency, reaper = self.reaper)
     enable_rolling_restart(rolling_restart_frequency) if rolling_restart_frequency
   end
 
-  def enable_rolling_restart(frequency = self.rolling_restart_frequency)
-    frequency = frequency + rand(0..10.0) # so all workers don't restart at the exact same time across multiple machines
-    AutoReap.new(frequency, RollingRestart.new).start
+  def enable_rolling_restart(frequency = self.rolling_restart_frequency,
+                             kill_delay = self.rolling_restart_kill_delay)
+    # so all workers don't restart at the exact same time across multiple machines
+    frequency = frequency + rand(0..10.0)
```

Comment:
It would also be good to configure this random function a bit. 10 seconds is still pretty coordinated for a large fleet. Normally I would say using a percent of the frequency. Maybe 10-20%.

Reply:
I made https://github.com/schneems/puma_worker_killer/pull/78 to help with this.